pkg/utils/kube/image.go

```diff
@@ -9,11 +9,32 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+type ImageExtractorConfigs map[string][]*ImageExtractorConfig
+
+type ImageExtractorConfig struct {
+	// Path is the path to the object containing the image field in a custom resource.
+	// It should be slash-separated. Each slash-separated key must by a valid YAML key or a wildcard '*'.
+	// Wildcard keys are expanded in case of arrays or objects.
+	Path string `json:"path" yaml:"path"`
+	// Value is an optional name of the field within 'path' that points to the image URI.
+	// This is useful when a custom 'key' is also defined.
+	// +optional
+	Value string `json:"value,omitempty" yaml:"value,omitempty"`
+	// Name is the entry the image will be available under 'images.<name>' in the context.
+	// If this field is not defined, image entries will appear under 'images.custom'.
+	// +optional
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
+	// Key is an optional name of the field within 'path' that will be used to uniquely identify an image.
+	// Note - this field MUST be unique.
+	// +optional
+	Key string `json:"key,omitempty" yaml:"key,omitempty"`
+}
+
```

Comment:
IMO we should keep the image verification specific types in [image_verification_types.go](https://github.com/kyverno/kyverno/blob/main/api/kyverno/v1/image_verification_types.go)

Reply:
It's not possible since the types package imports kubeutils so there ends up being a circular dependency.

Reply:
I can try and restructure some of this in a subsequent PR to avoid circular import. Didn't want to do it here since they would introduce unrelated changes to this one.

Reply:
Created https://github.com/kyverno/kyverno/issues/3600